v1/transform.go

```diff
@@ -0,0 +1,344 @@
+	valueMappings := make(map[string]string, len(mappings))
+	for k, v := range mappings {
+		valueMappings[v] = k
+	}
+	return valueMappings
+}
+
+func toV1CoreCommand(deviceCommands []models.DeviceCommand) []v1Models.Command {
+	commands := make([]v1Models.Command, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = v1Models.Command{
+			Name: c.Name,
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_R) {
+			commands[i].Get = toV1GetAction(c.Name, c.ResourceOperations)
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_W) {
+			commands[i].Put = toV1PutAction(c.Name, c.ResourceOperations)
+		}
+	}
+
+	return commands
+}
+
+func toV1GetAction(cmdName string, resourceOperations []models.ResourceOperation) v1Models.Get {
+	expectedValues := make([]string, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		expectedValues[i] = ro.DeviceResource
+	}
+	action := v1Models.Action{
+		Path: fmt.Sprintf("/api/v1/device/{deviceId}/%s", cmdName),
+		Responses: []v1Models.Response{
+			{
+				Code:           strconv.Itoa(http.StatusOK),
+				Description:    fmt.Sprintf("Issue the Get command %s", cmdName),
+				ExpectedValues: expectedValues,
+			}, {
+				Code:        strconv.Itoa(http.StatusInternalServerError),
+				Description: "internal server error",
+			},
+		},
+	}
+	return v1Models.Get{Action: action}
+}
+
+func toV1PutAction(cmdName string, resourceOperations []models.ResourceOperation) v1Models.Put {
+	parameterNames := make([]string, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		parameterNames[i] = ro.DeviceResource
+	}
+	action := v1Models.Action{
+		Path: fmt.Sprintf("/api/v1/device/{deviceId}/%s", cmdName),
+		Responses: []v1Models.Response{
+			{
+				Code:        strconv.Itoa(http.StatusNoContent),
+				Description: fmt.Sprintf("Issue the Put command %s", cmdName),
+			}, {
+				Code:        strconv.Itoa(http.StatusInternalServerError),
+				Description: "internal server error",
+			},
+		},
+	}
+	return v1Models.Put{
+		Action:         action,
+		ParameterNames: parameterNames,
+	}
+}
+
+// TransformToV2DeviceProfile transform v1 profile to v2
+func TransformToV2DeviceProfile(profile v1Models.DeviceProfile) models.DeviceProfile {
+	dto := models.DeviceProfile{
+		Description:  profile.Description,
+		Name:         profile.Name,
+		Manufacturer: profile.Manufacturer,
+		Model:        profile.Model,
+		Labels:       profile.Labels,
+	}
+	dto.DeviceResources = toV2DeviceResources(profile)
+	dto.DeviceCommands = toV2DeviceCommands(profile.DeviceCommands)
+	return dto
+}
+
+func toV2DeviceResources(profile v1Models.DeviceProfile) []models.DeviceResource {
+	resources := make([]models.DeviceResource, len(profile.DeviceResources))
+	for i, r := range profile.DeviceResources {
+		resources[i] = models.DeviceResource{
+			Description: r.Description,
+			Name:        r.Name,
+			IsHidden:    isResourceHidden(r.Name, profile),
```

Comment:
set the default is true and check CoreCommand later

Reply:
Modified as you suggest.